gems/smithy/lib/smithy/anvil/client/views/shapes.rb

```diff
@@ -0,0 +1,241 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Anvil
+    module Client
+      module Views
+        # @api private
+        class Shapes < View
+          def initialize(plan)
+            @plan = plan
+            @model = plan.model
+            @service_shape = assemble_service_shape
+            @shapes, @operation_shapes = assemble_shapes
+            super()
+          end
+
+          def namespace
+            Tools::Namespace.namespace_from_gem_name(@plan.options[:gem_name])
+          end
+
+          def shapes_with_members
+            @shapes.reject { |s| s.members.empty? }
+          end
+
+          private
+
+          def assemble_service_shape
+            shape = Vise::ServiceIndex.new(@model).service
+            shape_data = shape.values.first
+
+            ServiceShape.new(
+              id: shape.keys.first,
+              traits: filter_traits(shape_data['traits']),
+              version: shape_data['version']
+            )
+          end
+
+          def assemble_shapes
+            serializable_shapes = []
+            operation_shapes = []
+
+            @model['shapes'].each do |id, shape|
+              case shape['type']
+              when 'service', 'resource' then next
+              when 'operation'
+                operation_shapes << assemble_operation_shape(id, shape)
+              else
+                serializable_shapes << assemble_shape(id, shape)
```

Comment:
How do we know these are serializable?  I assume this is more about rendering the runtime model and not having checking traits to see if a shape is serializable or not.

Reply:
Yes, serializable is not good naming. They are just shapes to be rendered.

Reply:
Yeah I think my naming could be better here, will advise